ftplugin/sml.vim

```diff
@@ -13,24 +13,20 @@ setlocal iskeyword+='
 setlocal commentstring=(*%s*)
 
 " ----- Raimondi/delimitMate -----
-let b:delimitMate_quotes = "\""
+let b:delimitMate_quotes = '"'
 
 " ----- scrooloose/syntastic -----
 " Attempt to detect CM files in SML/NJ checker
-function! s:DetectCM(fname) abort
-  let cm = syntastic#util#findGlobInParent('*.cm', fnamemodify(a:fname, ':p:h'))
-  if cm !=# ''
-    let buf = bufnr(fnameescape(a:fname))
-    call setbufvar(buf, 'syntastic_sml_smlnj_args', '-m ' . syntastic#util#shescape(cm))
-    call setbufvar(buf, 'syntastic_sml_smlnj_fname', '')
-  endif
-endfunction
-
-call s:DetectCM(expand('<amatch>', 1))
```

Comment:
Your first version was done with an `autocmd`.  With an `autocmd` the `%` is not guaranteed to point to the file that triggered the `autocmd`, hence the dance with `<amatch>`.  It's a trick that's supposed to get you the right filename.  This used to be a major people in the past, as nobody expected `autocmd`s to behave that way.  The situation has improved over time (that is, `%` in `autocmd`s _usually_ points to the file that triggered the `autocmd` these days), but it isn't clear whether it was fixed for good or not.  There was no official statement from Bram either way.